crypto/dh/dh_gen.c

```diff
@@ -30,18 +30,27 @@ int DH_generate_parameters_ex(DH *ret, int prime_len, int generator,
 
 /*-
  * We generate DH parameters as follows
- * find a prime q which is prime_len/2 bits long.
- * p=(2*q)+1 or (p-1)/2 = q
+ * find a prime p which is prime_len bits long,
+ * where q=(p-1)/2 is also prime.
  * For this case, g is a generator if
  * g^((p-1)/q) mod p != 1 for values of q which are the factors of p-1.
  * Since the factors of p-1 are q and 2, we just need to check
  * g^2 mod p != 1 and g^q mod p != 1.
+ * Or in terms of the Legendre symbol: (g/p) == -1.
```

Comment:
This does not replace g^2 mod p != 1, or g != p-1,
since (-1/p) = -1 _if_ p mod 3 = 3.
Is that clear enough or should I mention g != p-1 and (g/p) == -1 ?